### DISCUSSION TITLE
Mutation page update without providing content

### DISCUSSION THREAD

**Discussion by IBRMHIRI** [requarks/wiki#5910] [Error / Bug Report] [answered]
It would be nice if one could just rename page (page title) or edit the path using the GraphQL API calls without providing the page content (which is the normal expect behaviour from GraphQL since only the id is required for the mutation update query).

I also tried to read out the content of a page using a get page query and use it again for my update query, however I get the following message:
```
{"data":{"pages":{"update":{"responseResult":{"message":"Cannot read properties of undefined (reading \'map\')"}}}}}
```
I assume that it is an internal bug, however it would be great if someone has a hint why I am getting this message 😸

**Comment by NGPixel** [comment, marked as answer]
The current GraphQL API is only meant to be used by the frontend. So that means you have to emulate its behavior, including fields that may not be required in the schema definition.

This is being addressed in the upcoming v3 so you'll be able to update only specific fields and the schema will be followed in a more standard way.
